enhancements/image-references.md

```diff
@@ -0,0 +1,241 @@
+process for creating bundle images, we must make sure a seamless migration is possible.
+
+### Risks and Mitigations
+
+It will be critical that the design be vetted by all of the concerned parties, from
+operator manifest authors to CI system authors and productized pipeline authors to
+ensure that the UX is appropriate in all cases. Furthermore, the largest risk in the
+implementation here is not prioritizing a clean migration pathway for all current
+users who create bundle images, which would lead to further fragmentation of the
+ecosystem, which is directly opposed to the goal of this enhancement.
+
+## Design Details
+
+The definition of a minimally-viable operator bundle image will be changed to
+ensure that all image references in the contained manifests have been resolved 
+to a digest and had the pull specifications rewritten to refer to those digests.
+
+The only acceptable process for creating an operator bundle will be to run the
+`operator-sdk generate bundle` CLI, providing the manifests, metadata and image
+sources as input to the creation process.
+
+### Proposed UX
+
+A set of new files in the `metadata/` directory of a bundle will be authored by
+the manifest authors and build systems to record image replacement intent and
+execution.
+
+Operator manifest authors write a manifest that refers to images using some
+opaque string, and provide an `image-references.yaml` file alongside their
+manifests that declares which strings inside of their manifest are referring
+to pull specifications of images and names each occurence.
+
+The `metadata/image-references.yaml` file holds data in the following format, mapping
+common names of container images to their string placeholders in the manifests:
+
+```yaml
+imageReferences:
+- name: common-name
+  substitute: registry.svc.ci.openshift.org/openshift:image
+```
+
+This mapping, therefore, defines what needs to be replaced in manifests when
+they are bundled and identifies each replacement with a name. When
+`operator-sdk generate bundle` runs, it will require as input a second mapping
+from those names to literal image pull specifications and will run the replacement.
+In this manner, the configuration provided by the manifest author remains static
+regardless of the eventual replacement that a build system will execute.
+
+The second mapping required at build-time, known as `image-replacements.yaml`, will take
+a similar form, mapping the common names of images to their explicitly resolved pull
+specifications. This file will have the following format:
+
+```yaml
+imagePullSpecs:
+- name: common-name
+  pullSpec: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:298d0b496f67a0ec97a37c6e930d76db9ae69ee6838570b0cd0c688f07f53780
+```
+
+`operator-sdk generate bundle` will use this chain of mapping to perform replacements
+in the manifests before creating a bundle image layer. The layer creation will also
```

Comment:
For devs running the command themselves, will there be an option to `--skip-image-replacements` ? I'm thinking about while an operator author is actively developing, maybe they do just want to have some tag references so they can push updates to the images without having to rebuild the bundle every time.